src/irr/video/COpenGLSpecializedShader.h

```diff
@@ -4,14 +4,21 @@
 #include "irr/asset/ICPUSpecializedShader.h"
 #include "irr/video/IGPUSpecializedShader.h"
 #include "COpenGLExtensionHandler.h"
+#include "irr/video/COpenGLShader.h"
 
-namespace irr { namespace video
+namespace irr {
+namespace asset
+{
+class IGLSLCompiler;
+}
+
+namespace video
 {
 
 class COpenGLSpecializedShader : public IGPUSpecializedShader
 {
 public:
-    COpenGLSpecializedShader(video::IVideoDriver* _driver, const asset::ICPUSpecializedShader* _cpushader);
+    COpenGLSpecializedShader(const video::IVideoDriver* _driver, const asset::ICPUBuffer* _spirv, const asset::ISpecializationInfo* _specInfo);
```

Comment:
why do you need the driver?

Reply:
actually just for GLSL version.. so might as well get rid of that and put uint32_t parameter instead
(plus GL calls, but those are statics in COpenGLExtensionHandler so driver not really needed)

Reply:
ok but what do you need the version for?

Reply:
resolved